azurerm/resource_arm_netapp_volume.go

```diff
@@ -0,0 +1,357 @@
+	accountName := id.Path["netAppAccounts"]
+	poolName := id.Path["capacityPools"]
+	name := id.Path["volumes"]
+
+	_, err = client.Delete(ctx, resourceGroup, accountName, poolName, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting NetApp Volume %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return waitForNetAppVolumeToBeDeleted(ctx, client, resourceGroup, accountName, poolName, name)
+}
+
+func waitForNetAppVolumeToBeDeleted(ctx context.Context, client *netapp.VolumesClient, resourceGroup, accountName, poolName, name string) error {
+	log.Printf("[DEBUG] Waiting for NetApp Volume Provisioning Service %q (Resource Group %q) to be deleted", name, resourceGroup)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"200", "202"},
+		Target:  []string{"404"},
+		Refresh: netappVolumeDeleteStateRefreshFunc(ctx, client, resourceGroup, accountName, poolName, name),
+		Timeout: 20 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for NetApp Volume Provisioning Service %q (Resource Group %q) to be deleted: %+v", name, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func netappVolumeDeleteStateRefreshFunc(ctx context.Context, client *netapp.VolumesClient, resourceGroupName string, accountName string, poolName string, name string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		res, err := client.Get(ctx, resourceGroupName, accountName, poolName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(res.Response) {
+				return nil, "", fmt.Errorf("Error retrieving NetApp Volume %q (Resource Group %q): %s", name, resourceGroupName, err)
+			}
+		}
+
+		if _, err := client.Delete(ctx, resourceGroupName, accountName, poolName, name); err != nil {
+			log.Printf("Error reissuing NetApp Volume %q delete request (Resource Group %q): %+v", name, resourceGroupName, err)
+		}
+
+		return res, strconv.Itoa(res.StatusCode), nil
+	}
+}
+
+func expandArmExportPolicyRule(input []interface{}) *netapp.VolumePropertiesExportPolicy {
+	results := make([]netapp.ExportPolicyRule, 0)
+	for _, item := range input {
+		if item != nil {
+			v := item.(map[string]interface{})
+			ruleIndex := int32(v["rule_index"].(int))
+			allowedClients := v["allowed_clients"].(string)
+			cifs := v["cifs"].(bool)
+			nfsv3 := v["nfsv3"].(bool)
+			nfsv4 := v["nfsv4"].(bool)
+			unixReadOnly := v["unix_read_only"].(bool)
+			unixReadWrite := v["unix_read_write"].(bool)
+
+			result := netapp.ExportPolicyRule{
+				AllowedClients: utils.String(allowedClients),
+				Cifs:           utils.Bool(cifs),
+				Nfsv3:          utils.Bool(nfsv3),
+				Nfsv4:          utils.Bool(nfsv4),
+				RuleIndex:      utils.Int32(ruleIndex),
+				UnixReadOnly:   utils.Bool(unixReadOnly),
+				UnixReadWrite:  utils.Bool(unixReadWrite),
+			}
+
+			results = append(results, result)
+		}
+	}
+
+	result := netapp.VolumePropertiesExportPolicy{
+		Rules: &results,
+	}
```

Comment:
we may as well return this directly here, rather than assigning it to a variable

Reply:
fixed